### PR TITLE
Version 49

### PR DESCRIPTION
**Pull request by vinniefalco**
This includes version 48

* Use <iosfwd> instead of <ostream>

HTTP:

* Add HEAD request example

API Changes:

* Refactor method and verb
* Canonicalize string_view parameter types
* Tidy up empty_body writer error
* Refactor header status, reason, and target

Documentation preview:
http://vinniefalco.github.io/stage/beast/v49

### CODE REVIEW

File: doc/6_1_streams.qbk [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+[note
+    Code which declares stream objects using Asio SSL types must
+    toinclude the file `<beast/websocket/ssl.hpp>`.
+]
+
+[heading Non-owning References]
+
+If a socket type supports move construction, a websocket stream may be
+constructed around the already existing socket by invoke the move
```
Comment (nbougalis) [inline doc/6_1_streams.qbk]: "by invoke" should be "by invoking" or, better yet:

File: doc/6_6_control.qbk [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+```
+    void on_ping(bool is_pong, websocket::ping_data const& payload);
+    ...
+    ws.set_option(ping_callback{&on_ping});
+```
+
+When a ping callback is registered, all pings and pongs received through
+either synchronous read functions or asynchronous read functions will
+invoke the ping callback, with the value of `is_pong` set to `true` if a
+pong was received else `false` if a ping was received. The payload of
```
Comment (nbougalis) [inline doc/6_6_control.qbk]: 'else `false' if a ping was received.` -> 'or `false` otherwise.'

File: doc/8_0_design.qbk [hunk 3]
```diff
@@ -7,15 +7,6 @@
...
-<informaltable frame="all"><tgroup cols="1"><colspec colname="a"/><tbody><row><entry valign="top"><simplelist>
-  <member><link linkend="beast.design.http_message">HTTP Message Container</link></member>
-  <member><link linkend="beast.design.http_comparison">HTTP Comparison to Other Libraries</link></member>
-  <member><link linkend="beast.design.websocket_zaphoyd">Comparison to Zaphoyd Studios WebSocket++</link></member>
-  <member><link linkend="beast.design.review">Boost Formal Review FAQ</link></member>
-</simplelist></entry></row></tbody></tgroup></informaltable>
-''']
-
 The implementations are driven by business needs of cryptocurrency server
 applications (e.g. [@https://ripple.com Ripple]) written in C++. These
```
Comment (nbougalis) [inline doc/8_0_design.qbk]: >applications (e.g. [@https://ripple.com Ripple]) written in C++.

should be:

>applications, such as [@https://github.com/ripple/rippled rippled], written in C++. 

Send people to the `rippled` github ;)

File: doc/8_0_design.qbk [hunk 4]
```diff
@@ -7,15 +7,6 @@
...
-[block '''
-<informaltable frame="all"><tgroup cols="1"><colspec colname="a"/><tbody><row><entry valign="top"><simplelist>
-  <member><link linkend="beast.design.http_message">HTTP Message Container</link></member>
-  <member><link linkend="beast.design.http_comparison">HTTP Comparison to Other Libraries</link></member>
-  <member><link linkend="beast.design.websocket_zaphoyd">Comparison to Zaphoyd Studios WebSocket++</link></member>
-  <member><link linkend="beast.design.review">Boost Formal Review FAQ</link></member>
-</simplelist></entry></row></tbody></tgroup></informaltable>
-''']
-
 The implementations are driven by business needs of cryptocurrency server
```
Comment (nbougalis) [inline doc/8_0_design.qbk]: "were originally driven by"

File: doc/8_0_design.qbk (hunk 3 above)
Comment (nbougalis) [inline doc/8_0_design.qbk]: Also, lines 24-25:

>Leave important decisions to the user, such as allocating memory or
>managing flow control.

Should probably be:

>Leave important decisions, such as allocating memory or
>managing flow control, to to the user.

File: doc/6_6_control.qbk (hunk 2 above)
Comment (vinniefalco) [inline doc/6_6_control.qbk]: great change

File: doc/8_0_design.qbk (hunk 3 above)
Comment (vinniefalco) [inline doc/8_0_design.qbk]: wow strong deja-vu there lol

File: doc/1_overview.qbk [hunk 5]
```diff
@@ -7,10 +7,19 @@
...
+    and WebSocket protocol programming that uses the consistent network and
+    asynchronous model of __Asio__.
+    
+    Beast is not an HTTP client or HTTP server, but it can be used to
+    build those things. The library is intended to be a foundation upon
+    which new libraries may be built. It is a goal that other architects
+    will create interoperable solutions on top of Beast. The provided
+    example programs shows how a client or server might be built.
+]
+
```
Comment (vinniefalco) [inline doc/1_overview.qbk]: willfix

Comment (vinniefalco) [inline doc/1_overview.qbk]: willfix

File: doc/1_overview.qbk [hunk 6]
```diff
@@ -118,7 +123,7 @@ and
 [@https://www.ripple.com Ripple Labs]
 for supporting its early development. Also thanks to
```
Comment (vinniefalco) [inline doc/1_overview.qbk]: willfix

File: doc/3_3_buffers.qbk [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+]]
+]
+
+Network applications frequently need to manipulate buffer sequences. To
+facilitate working with buffers the library treats these sequences as
+a special type of range. Algorithms and wrappers are provided which
+transform these buffer sequences ranges efficiently using lazy evaluation.
+No memory allocations are used in the transformations; instead, they
+create lightweight iterators over the existing, unmodified memory
+buffers. The lifetime of the transformed buffers is retained by the
```
Comment (vinniefalco) [inline doc/3_3_buffers.qbk]: *"Control of buffers is retained by the caller; ownership is not transferred."*

Comment (vinniefalco) [inline doc/3_3_buffers.qbk]: willfix

File: doc/3_3_buffers.qbk [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    [link beast.ref.buffer_cat `buffer_cat`]
+][
+    This functions returns a new buffer sequence which, when iterated,
+    traverses the sequence which would be formed if all of the input buffer
+    sequences were concatenated. With this routine, multiple calls to a
+    stream's `write_some` function may be combined into one, eliminating
+    expensive system calls.
+]]
+[[
+    [link beast.ref.buffer_cat_view `buffer_cat_view`]
```
Comment (vinniefalco) [inline doc/3_3_buffers.qbk]: Yeah I know but its the lesser of the evils

File: doc/6_1_streams.qbk (hunk 1 above)
Comment (vinniefalco) [inline doc/6_1_streams.qbk]: willfix

File: doc/8_0_design.qbk (hunk 4 above)
Comment (vinniefalco) [inline doc/8_0_design.qbk]: willfix

File: doc/6_6_control.qbk (hunk 2 above)
Comment (vinniefalco) [inline doc/6_6_control.qbk]: sketchy but I will change it anyway.

File: doc/8_0_design.qbk (hunk 3 above)
Comment (vinniefalco) [inline doc/8_0_design.qbk]: willfix

**Comment by vinniefalco** [comment]
Thanks

**Pull request closed without merging**